tests/Metalama.Framework.Tests.Integration/Tests/Aspects/Misc/Issue31159.cs

```diff
@@ -1,23 +1,31 @@
 using System;
 using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+using Metalama.Framework.Eligibility;
 
 namespace Metalama.Framework.Tests.Integration.Tests.Aspects.Misc.Issue31159;
 
 [RunTimeOrCompileTime]
 public class DerivedAspect : BaseAspect
 {
+    public override void BuildEligibility( IEligibilityBuilder<IParameter> builder )
+    {
```

Comment:
Applying a contract to an abstract method is allowed but only as Abstract, i.e. the rule should use the `ExceptForInheritance` prefix.

Reply:
I.e. it should work and if it does not, there is another bug.